modular_skyrat/modules/fauna_reagent/fauna_reagent.dm

```diff
@@ -0,0 +1,28 @@
+/mob/living/simple_animal
+	/// whether the simple animal can be healed/damaged through reagents
+	var/reagent_health = FALSE
+
+/mob/living/simple_animal/Initialize(mapload)
+	. = ..()
+	if(reagent_health)
+		create_reagents(1000, REAGENT_HOLDER_ALIVE)
+
+/mob/living/simple_animal/hostile/megafauna
+	reagent_health = TRUE
+
+/mob/living/simple_animal/hostile/asteroid
+	reagent_health = TRUE
+
+/mob/living/simple_animal/Life(delta_time, times_fired)
+	. = ..()
+	if(reagent_health && reagents)
+		for(var/datum/reagent/reagents_within in reagents.reagent_list)
```

Comment:
```suggestion
		for(var/datum/reagent/reagents_within as anything in reagents.reagent_list)
```

Reply:
how it is currently written works. I can test later if this is fine as well.

Reply:
above method is faster